Src/main.c

```diff
@@ -101,13 +101,25 @@ int main(void)
   debug("Compiled on %s at %s", __DATE__, __TIME__);
 
 
+
   HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, GPIO_PIN_SET);
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+  const uint32_t PWM_FREQUENCY = 50;
+  const uint32_t PWM_STEPS = 100;
+  const uint32_t COUNTER_FREQUENCY = PWM_FREQUENCY* PWM_STEPS;
+  const uint32_t TIMER_FREQUENCY = 48e6;
+  const uint32_t PRESCALER = (TIMER_FREQUENCY/COUNTER_FREQUENCY) - 1;
+  const uint32_t PERIOD = 2*1000; //ms * 1000 to get us
   while (1)
   {
+    uint32_t adcVal;
+    adcVal = HAL_ADC_GetValue(&hadc); //value between 0 and 4095, inclusive because its a 12 bit adc -> 4096
```

Comment:
It's best to use the lowest datatype size whenever possible... memory ain't free :)